drivers/sensor/lsm303dlhc_accel/lsm303dlhc_accel.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2018 Philémon Jaermann
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __SENSOR_LSM303DLHC_ACCEL_
+#define __SENSOR_LSM303DLHC_ACCEL_
+
+#define LSM303DLHC_ACCEL_I2C_ADDRESS	CONFIG_LSM303DLHC_ACCEL_I2C_ADDR
+
+#define LSM303_DLHC_ACCEL_X_EN_BIT	BIT(0)
+#define LSM303DLHC_ACCEL_Y_EN_BIT	BIT(1)
+#define LSM303DLHC_ACCEL_Z_EN_BIT	BIT(2)
+#define LSM303DLHC_ACCEL_EN_BITS	(LSM303_DLHC_ACCEL_X_EN_BIT | \
+					LSM303DLHC_ACCEL_Y_EN_BIT | \
+					LSM303DLHC_ACCEL_Z_EN_BIT)
+
+#if defined(CONFIG_LSM303_POWER_MODE_LOW)
+	#define LSM303DLHC_LP_EN_BIT	BIT(3)
+#elif defined(CONFIG_LSM303DLHC_POWER_MODE_NORMAL)
+	#define LSM303DLHC_LP_EN_BIT	0
+#endif
+
+#if defined(CONFIG_LSM303DLHC_ACCEL_ODR_1)
+	#define LSM303DLHC_ACCEL_ODR_IDX	1
+#elif defined(CONFIG_LSM303DLHC_ACCEL_ODR_2)
+	#define LSM303DLHC_ACCEL_ODR_IDX	2
+#elif defined(CONFIG_LSM303DLHC_ACCEL_ODR_3)
+	#define LSM303DLHC_ACCEL_ODR_IDX	3
+#elif defined(CONFIG_LSM303DLHC_ACCEL_ODR_4)
+	#define LSM303DLHC_ACCEL_ODR_IDX	4
+#elif defined(CONFIG_LSM303DLHC_ACCEL_ODR_5)
+	#define LSM303DLHC_ACCEL_ODR_IDX	5
+#elif defined(CONFIG_LSM303DLHC_ACCEL_ODR_6)
+	#define LSM303DLHC_ACCEL_ODR_IDX	6
+#elif defined(CONFIG_LSM303DLHC_ACCEL_ODR_7)
+	#define LSM303DLHC_ACCEL_ODR_IDX	7
+#elif defined(CONFIG_LSM303DLHC_ACCEL_ODR_8)
+	#define LSM303DLHC_ACCEL_ODR_IDX	8
+#elif defined(CONFIG_LSM303DLHC_ACCEL_ODR_9_NORMAL) || \
+	defined(CONFIG_LSM303DLHC_ACCEL_ODR_9_LOW)
+	#define LSM303DLHC_ACCEL_ODR_IDX	9
+#endif
+
+#define LSM303DLHC_ACCEL_ODR_SHIFT	4
+#define LSM303DLHC_ACCEL_ODR_BITS	(LSM303DLHC_ACCEL_ODR_IDX << \
+					LSM303DLHC_ACCEL_ODR_SHIFT)
+
+#if	defined(CONFIG_LSM303DLHC_ACCEL_RANGE_2G)
+	#define LSM303DLHC_ACCEL_FS_IDX	0
+#elif defined(CONFIG_LSM303DLHC_ACCEL_RANGE_4G)
+	#define LSM303DLHC_ACCEL_FS_IDX	1
+#elif defined(CONFIG_LSM303DLHC_ACCEL_RANGE_8G)
+	#define LSM303DLHC_ACCEL_FS_IDX	2
+#elif defined(CONFIG_LSM303DLHC_ACCEL_RANGE_16G)
+	#define LSM303DLHC_ACCEL_FS_IDX	3
+#endif
+
+#define LSM303DLHC_ACCEL_AUTO_INC_ADDR	BIT(7)
+#define LSM303DLHC_ACCEL_FS_SHIFT		4
+#define LSM303DLHC_ACCEL_FS_BITS		(LSM303DLHC_ACCEL_FS_IDX << \
+						LSM303DLHC_ACCEL_FS_SHIFT)
+
+#define LSM303DLHC_REG_CTRL_1	0x20
+#define LSM303DLHC_REG_CTRL_4	0x23
+#define LSM303DLHC_REG_ACCEL_X_LSB	(0x28 | LSM303DLHC_ACCEL_AUTO_INC_ADDR)
+#define LSM303DLHC_ACCEL_SCALE		(2 << LSM303DLHC_ACCEL_FS_IDX)
```

Comment:
I think you should multiply by SENSOR_G (like the lis3dh does) to obtain m/s2.
Now I see values in G when I use this driver (as I already commented I'd be in favour to re-use the lis3dh driver for the
accelerometer of the 303dlhc, as they are basically the same sensor).